### ISSUE TITLE
Import running `feast apply` with a fresh `feast[gcp]` install.

### ISSUE THREAD

**Issue by kylegallatin** [feast-dev/feast#3853]
## Expected Behavior 

Creates required infrastructure. 

## Current Behavior

Returns a large import error that results from trying to import ARROW_SCALAR_IDS_TO_BQ from google.cloud.bigquery._pandas_helpers.

```
/usr/local/anaconda3/lib/python3.8/site-packages/pandas/core/computation/expressions.py:20: UserWarning: Pandas requires version '2.7.3' or newer of 'numexpr' (version '2.7.1' currently installed).
  from pandas.core.computation.check import NUMEXPR_INSTALLED
<frozen importlib._bootstrap>:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 216 from C header, got 232 from PyObject
<frozen importlib._bootstrap>:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 216 from C header, got 232 from PyObject
Traceback (most recent call last):
  File "/usr/local/anaconda3/lib/python3.8/site-packages/feast/infra/offline_stores/bigquery.py", line 62, in <module>
    from google.cloud.bigquery._pandas_helpers import ARROW_SCALAR_IDS_TO_BQ
ImportError: cannot import name 'ARROW_SCALAR_IDS_TO_BQ' from 'google.cloud.bigquery._pandas_helpers' (/usr/local/anaconda3/lib/python3.8/site-packages/google/cloud/bigquery/_pandas_helpers.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/anaconda3/lib/python3.8/site-packages/feast/importer.py", line 26, in import_class
    module = importlib.import_module(module_name)
  File "/usr/local/anaconda3/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/usr/local/anaconda3/lib/python3.8/site-packages/feast/infra/offline_stores/bigquery.py", line 68, in <module>
    raise FeastExtrasDependencyImportError("gcp", str(e))
feast.errors.FeastExtrasDependencyImportError: cannot import name 'ARROW_SCALAR_IDS_TO_BQ' from 'google.cloud.bigquery._pandas_helpers' (/usr/local/anaconda3/lib/python3.8/site-packages/google/cloud/bigquery/_pandas_helpers.py)
You may need run pip install 'feast[gcp]'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/anaconda3/bin/feast", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/anaconda3/lib/python3.8/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/anaconda3/lib/python3.8/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/usr/local/anaconda3/lib/python3.8/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/anaconda3/lib/python3.8/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/anaconda3/lib/python3.8/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/anaconda3/lib/python3.8/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/anaconda3/lib/python3.8/site-packages/feast/cli.py", line 488, in apply_total_command
    repo_config = load_repo_config(repo, fs_yaml_file)
  File "/usr/local/anaconda3/lib/python3.8/site-packages/feast/repo_config.py", line 549, in load_repo_config
    c = RepoConfig(**raw_config)
  File "/usr/local/anaconda3/lib/python3.8/site-packages/feast/repo_config.py", line 185, in __init__
    super().__init__(**data)
  File "pydantic/main.py", line 339, in pydantic.main.BaseModel.__init__
  File "pydantic/main.py", line 1050, in pydantic.main.validate_model
  File "/usr/local/anaconda3/lib/python3.8/site-packages/feast/repo_config.py", line 385, in _validate_offline_store_config
    offline_config_class = get_offline_config_from_type(offline_store_type)
  File "/usr/local/anaconda3/lib/python3.8/site-packages/feast/repo_config.py", line 530, in get_offline_config_from_type
    return import_class(module_name, config_class_name, config_class_name)
  File "/usr/local/anaconda3/lib/python3.8/site-packages/feast/importer.py", line 31, in import_class
    raise FeastModuleImportError(module_name, class_name) from e
feast.errors.FeastModuleImportError: Could not import module 'feast.infra.offline_stores.bigquery' while attempting to load class 'BigQueryOfflineStoreConfig'
```

## Steps to reproduce

```shell
# in a fresh python env (tried both Google Cloud Shell and Anaconda)
conda create -n py39 python=3.9
conda activate py39
python -m pip install 'feast[gcp]'
feast init my_project -t gcp
feast -c my_project/feature_repo apply
```

### Specifications

- Version: 0.34.1
- Platform: Linux and MacOS
- Subsystem:

## Possible Solution
Guessing it might be the version of google.cloud.bigquery

**Comment by shuchu** [comment]
From the error: 
`feast.errors.FeastExtrasDependencyImportError: cannot import name 'ARROW_SCALAR_IDS_TO_BQ' from 'google.cloud.bigquery._pandas_helpers' (/usr/local/anaconda3/lib/python3.8/site-packages/google/cloud/bigquery/_pandas_helpers.py)`

It seems you are planning to use Python 3.9, but the error shows you are calling from python3.8.  do you mind to check the python env and run it again?

**Comment by kylegallatin** [comment]
Hey @shuchu! Thanks for the reply. I've confirmed the error from both 3.8 and 3.9, I was testing with both versions to make sure it wasn't a Python issue and pasted the wrong one.

**Comment by shuchu** [comment]
ok, I will check this. pls wait.

**Comment by kylegallatin** [comment]
Looks like `ARROW_SCALAR_IDS_TO_BQ` was moved to private `_ARROW_SCALAR_IDS_TO_BQ` as part of [this commit](https://github.com/googleapis/python-bigquery/commit/f637e5e38bc6c8d45ac83eeeaf4a59d7ac6b5d49). Seems to be part of release 3.13.0? 

`python3 -m pip install google-cloud-bigquery==3.12.0` resolves this issue.

**Comment by kylegallatin** [comment]
Ah - looks like this was recently resolved here: https://github.com/feast-dev/feast/commit/2151c39d1a8d8eba114306411dd4bd91ac0ce3f6

**Comment by sudohainguyen** [comment]
this is resolved in the latest release v0.35.0

**Comment by redhatHameed** [comment]
@kylegallatin can you close if it fixed.

cc: 
@jeremyary @etirelli
